torch/lib/c10d/reducer.cpp

```diff
@@ -472,6 +475,17 @@ void Reducer::mark_variable_ready(VariableIndex index) {
   TORCH_CHECK(
       variable_index < variable_locators_.size(),
       "Out of range variable index.");
+  if (ddp_debug_level_ != DistributedDebugLevel::OFF && replica_index == 0) {
```

Comment:
feel like perIterationReadyParams_ is useful even in default mode, where we can give the unready parameter's index in the error message

Reply:
Will update so that we always track/check this even in default mode.